internal/components/store/store.go

```diff
@@ -0,0 +1,19 @@
+package store
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type UniqueObject interface {
+	GetIdentifier() types.UID
```

Comment:
If this is a k8s UID, not a Grafana UID, then it's not the one we'd want to expose for typical db querying purposes, right? per https://docs.google.com/document/d/1Y_HaKMIfJxg1ITdOStdQU3PanaU9FqWBsB-uQbU-5a0/edit#heading=h.abjojhgl6idr